ouroboros-network/src/Ouroboros/Network/Protocol/ChainSync/Codec.hs

```diff
@@ -11,35 +11,41 @@ module Ouroboros.Network.Protocol.ChainSync.Codec
   , codecChainSyncId
   ) where
 
+import           Control.Monad (when)
 import           Control.Monad.Class.MonadST
 
 import           Network.TypedProtocol.Codec
 import           Ouroboros.Network.Codec
 import           Ouroboros.Network.Protocol.ChainSync.Type
 
+import qualified Data.ByteString      as BS
+import qualified Data.ByteString.Lazy as LBS
 
 import qualified Codec.CBOR.Encoding as CBOR
-import qualified Codec.CBOR.Read     as CBOR
 import qualified Codec.CBOR.Decoding as CBOR
+import           Codec.CBOR.Encoding (encodeListLen, encodeWord)
+import           Codec.CBOR.Decoding (decodeListLen, decodeWord)
+import qualified Codec.CBOR.Read     as CBOR
+import qualified Codec.CBOR.Write    as CBOR
 
-import Data.ByteString.Lazy (ByteString)
-import Codec.CBOR.Encoding (encodeListLen, encodeWord)
-import Codec.CBOR.Decoding (decodeListLen, decodeWord)
 
 -- | The main CBOR 'Codec' for the 'ChainSync' protocol.
 --
 codecChainSync :: forall header point m.
                   (MonadST m)
                => (header -> CBOR.Encoding)
```

Comment:
The byron-proxy codec *could* work if this is changed to `(header -> ByteString)`, although I'd recommend going back to the more general treatment where the user chooses CBOR-in-CBOR if wanted/needed.